code/modules/projectiles/gun_system.dm

```diff
@@ -814,7 +814,7 @@ and you're good to go.
 	if((flags_gun_features & GUN_WIELDED_FIRING_ONLY) && !(flags_item & WIELDED)) //If we're not holding the weapon with both hands when we should.
 		to_chat(user, "<span class='warning'>You need a more secure grip to fire this weapon!")
 		return FALSE
-	if(user.do_actions)
+	if(LAZYACCESS(user.do_actions, src))
```

Comment:
wielding action is not from src btw ?

Reply:
From what I understand, the thing that is added to do_actions is a reference of the target. The target in do_wield's do_mob is the holder of the gun, while the target of the do_after in toggle_aim_mode is the gun itself.